python/foglamp/services/south/server.py

```diff
@@ -210,19 +181,40 @@ def run(self):
         for signal_name in (signal.SIGINT, signal.SIGTERM):
             loop.add_signal_handler(
                 signal_name,
-                lambda: asyncio.ensure_future(self._stop(loop)))
+                lambda: asyncio.ensure_future(self.stop()))
 
         asyncio.ensure_future(self._start(loop))
         loop.run_forever()
 
-    async def shutdown(self, request):
+    async def _stop(self):
+        if self._plugin is not None:
+            try:
+                self._plugin.plugin_shutdown(self._plugin_handle)
+            except Exception:
+                _LOGGER.exception("Unable to stop plugin '{}'".format(self._name))
+            finally:
+                self._plugin = None
+                self._plugin_handle = None
+
+        try:
+            await Ingest.stop()
+        except Exception:
+            _LOGGER.exception('Unable to stop the Ingest server')
+            return
+
+        # Stop all pending asyncio tasks
+        for task in asyncio.Task.all_tasks():
+            task.cancel()
+
+    async def stop(self, request=None):
         """implementation of abstract method form foglamp.common.microservice.
         """
-        print("shutdown south")
-        return web.json_response({"south":"shutdown"})
+        await self._stop()
+        asyncio.get_event_loop().stop()
+        _LOGGER.info('Stopping South Service PLugin {}'.format(self._name))
 
-    async def change(self, request):
+    async def change(self, request=None):
```

Comment:
not none; it will only be called by microservice mgt api client to notify category (registered interest) change